mbed/mbed.py

```diff
@@ -1447,6 +1440,22 @@ def ignore_build_dir(self):
             except IOError:
                 error("Unable to write build ignore file in \"%s\"" % os.path.join(build_path, '.mbedignore'), 1)
 
+    def detect_target(self, info=None):
```

Comment:
Say I have a LPC1768, a K64F, and an NRF51_DK plugged into my machine. I could run `mbed compile -m NRF51_DK ... --flash` and mbed CLI could correctly figure out which board to flash. Can be useful when developing for multiple platforms, but maybe not important as of right now.

> Also what if I have multiple targets of the same kind mixed with different ones?

In this case it would have to error like you do above since there are multiple boards (for example, 2 LPC1768s plugged in. You don't know which one to flash just by using the target name).

I'm afraid I'm hindering this PR more than I'm helping it with this comment. I'm happy to submit the change I mentioned above as a PR after this is merged for more discussion.